ntypes error

I set the typemap in param.json as ["Co","Ni"], and use a dataset of bulk Ni. Then ntypes error happens！
```
Traceback (most recent call last):
  File "/nfs-share/home/1800011848/miniconda3/bin/dp", line 10, in <module>
    sys.exit(main())
  File "/nfs-share/home/1800011848/miniconda3/lib/python3.8/site-packages/deepmd/main.py", line 66, in main
    train(args)
  File "/nfs-share/home/1800011848/miniconda3/lib/python3.8/site-packages/deepmd/train.py", line 81, in train
    _do_work(jdata, run_opt)
  File "/nfs-share/home/1800011848/miniconda3/lib/python3.8/site-packages/deepmd/train.py", line 134, in _do_work
    model.build (data, stop_batch)
  File "/nfs-share/home/1800011848/miniconda3/lib/python3.8/site-packages/deepmd/Trainer.py", line 209, in build
    assert (self.ntypes == data.get_ntypes()), "ntypes should match that found in data"
AssertionError: ntypes should match that found in data
```
And the same param works well when using a alloy dataset( all type.raw have two elements)

--- a/source/train/Trainer.py
+++ b/source/train/Trainer.py
@@ -209,7 +209,10 @@ class NNPTrainer (object):
                data, 
                stop_batch = 0) :
         self.ntypes = self.model.get_ntypes()
-        assert (self.ntypes == data.get_ntypes()), "ntypes should match that found in data"
+        # Usually, the type number of the model should be equal to that of the data
+        # However, nt_model > nt_data should be allowed, since users may only want to 
+        # train using a dataset that only have some of elements 
+        assert (self.ntypes >= data.get_ntypes()), "ntypes should match that found in data"
         self.stop_batch = stop_batch
 
         self.batch_size = data.get_batch_size()

